variables.tf

```diff
@@ -118,3 +118,18 @@ variable "cors_rule" {
   type        = any
   default     = []
 }
+
+variable "event-notification-config" {
+  type = object({
+    target_type   = string,
+    queue_name    = string,
+    filter_prefix = string,
+    events        = optional(list(string), ["s3:ObjectCreated:*"])
+  })
+  default = {
+    target_type   = "null"
+    queue_name    = "test"
+    filter_prefix = "test/"
+    events        = ["s3:ObjectCreated:*"]
+  }
+}
```

Comment:
we use variable names in snake_case format, can you use `event_notification_config`?
please have descripotion about this variable and its fields, also some info about possible values for some fields, like target_type ?
the queue name seems can be created based on bucket name and instead of `queue_name` to have some general field for addintin unique suffix to name of queue, so that in case of other type of targets we can use it also. Something like have `name_suffix` field in place with default enpty string and  have queue named in following form: queue_name = `${var.name}${var.event_notification_config.name_suffix}` ?